internal/core/src/index/ScalarIndexSort-inl.h

```diff
@@ -307,4 +313,46 @@ ScalarIndexSort<T>::Reverse_Lookup(size_t idx) const {
     auto offset = idx_to_offsets_[idx];
     return data_[offset].a_;
 }
+
+template <typename T>
+inline bool
+ScalarIndexSort<T>::outOfRange(const T value, const OpType op) {
+    if (!data_.empty()) {
+        auto lower = data_.begin();
+        auto upper = data_.rbegin();
+        bool outOfRange = false;
+        switch (op) {
+            case OpType::LessThan:
+                outOfRange = lower->a_ >= value;
+                break;
+            case OpType::LessEqual:
+                outOfRange = lower->a_ > value;
+                break;
+            case OpType::GreaterThan:
+                outOfRange = upper->a_ <= value;
+                break;
+            case OpType::GreaterEqual:
+                outOfRange = upper->a_ < value;
+                break;
+            default:
+                throw SegcoreError(OpTypeInvalid,
+                                   fmt::format("Invalid OperatorType: {}", op));
```

Comment:
error message is not enough exact, just InvalidType for check index optimization